### PR TITLE
FINERACT-1895 - Fixing the issue where Fineract doesn't start with spring profile activeMqEnabled

### PR DESCRIPTION
**Pull request by hiteshprajapati**
Using correct Listener for ActiveMQ
adding missing property fineract.mode.batch-enabled

## Description

Describe the changes made and why they were made.

Ignore if these details are present on the associated [Apache Fineract JIRA ticket](https://github.com/apache/fineract/pull/1284).

## Checklist

Please make sure these boxes are checked before submitting your pull request - thanks!

- [ ] Write the commit message as per https://github.com/apache/fineract/#pull-requests

- [ ] Acknowledge that we will not review PRs that are not passing the build _("green")_ - it is your responsibility to get a proposed PR to pass the build, not primarily the project's maintainers.

- [ ] Create/update unit or integration tests for verifying the changes made.

- [ ] Follow coding conventions at https://cwiki.apache.org/confluence/display/FINERACT/Coding+Conventions.

- [ ] Add required Swagger annotation and update API documentation at fineract-provider/src/main/resources/static/legacy-docs/apiLive.htm with details of any API changes

- [ ] Submission is not a "code dump".  (Large changes can be made "in repository" via a branch.  Ask on the developer mailing list for guidance, if required.)

FYI our guidelines for code reviews are at https://cwiki.apache.org/confluence/display/FINERACT/Code+Review+Guide.

### CODE REVIEW

File: fineract-provider/src/main/resources/application.properties [hunk 1]
```diff
@@ -37,6 +37,7 @@ fineract.tenant.description=${FINERACT_DEFAULT_TENANTDB_DESCRIPTION:Default Demo
 
 fineract.mode.read-enabled=${FINERACT_MODE_READ_ENABLED:true}
 fineract.mode.write-enabled=${FINERACT_MODE_WRITE_ENABLED:true}
+fineract.mode.batch-enabled=${FINERACT_MODE_BATCH_ENABLED:true}
```
Comment (galovics) [inline fineract-provider/src/main/resources/application.properties:40]: There's no instance mode like this anymore, why was this required?

File: fineract-provider/src/main/java/org/apache/fineract/notification/config/MessagingConfiguration.java [hunk 2]
```diff
@@ -44,7 +44,7 @@ public class MessagingConfiguration {
     private Environment env;
 
     @Autowired
-    private NotificationEventListener notificationEventListener;
+    private ActiveMQNotificationEventListener notificationEventListener;
```
Comment (galovics) [inline fineract-provider/src/main/java/org/apache/fineract/notification/config/MessagingConfiguration.java:47]: This is fine but I'd love to get rid of the spring profile for this functionality and replace it with proper env variable configuration option.

This was rather a note and not a requirement from my side to get this PR in.

**Review by galovics — CHANGES_REQUESTED** [review]

File: fineract-provider/src/main/resources/application.properties (hunk 1 above)
Comment (hiteshprajapati) [reply fineract-provider/src/main/resources/application.properties:40]: @galovics This instance mode is used in two conditions - EnableFineractEventsCondition and EnableFineractEventListenerCondition

Comment (hiteshprajapati) [reply fineract-provider/src/main/resources/application.properties:40]: @galovics : This particular mode is used in two conditions - EnableFineractEventListenerCondition and EnableFineractEventsCondition

And I see the same mode has unit tests written for it. And thats reason the mode exists in application-test.properties file

File: fineract-provider/src/main/java/org/apache/fineract/notification/config/MessagingConfiguration.java (hunk 2 above)
Comment (hiteshprajapati) [reply fineract-provider/src/main/java/org/apache/fineract/notification/config/MessagingConfiguration.java:47]: I understood

**Comment by hiteshprajapati** [comment]
@galovics @vidakovic The failed checks run successfully in my fork. So above failure looks like random failure due to API call.

https://github.com/hiteshprajapati/fineract/actions/runs/4216892567
https://github.com/hiteshprajapati/fineract/actions/runs/4216892551

File: fineract-provider/src/main/resources/application.properties (hunk 1 above)
Comment (hiteshprajapati) [reply fineract-provider/src/main/resources/application.properties:40]: @galovics Please provide your feedback.
